tests/func/test_arithmetic.cpp

```diff
@@ -282,40 +282,40 @@ TEST(Arithmetic, OptionArithmeticInt32)
 {
   nd::array NA = nd::empty(ndt::type("?int32"));
   nd::assign_na(NA);
-  EXPECT_FALSE(nd::is_avail(NA + 1));
-  EXPECT_FALSE(nd::is_avail(NA - 1));
-  EXPECT_FALSE(nd::is_avail(NA * 1));
-  EXPECT_FALSE(nd::is_avail(NA / 1));
-
-  EXPECT_FALSE(nd::is_avail(1 + NA));
-  EXPECT_FALSE(nd::is_avail(1 - NA));
-  EXPECT_FALSE(nd::is_avail(1 * NA));
-  EXPECT_FALSE(nd::is_avail(1 / NA));
-
-  EXPECT_FALSE(nd::is_avail(NA + NA));
-  EXPECT_FALSE(nd::is_avail(NA - NA));
-  EXPECT_FALSE(nd::is_avail(NA * NA));
-  EXPECT_FALSE(nd::is_avail(NA / NA));
+  EXPECT_ALL_FALSE(nd::is_avail(NA + 1));
+  EXPECT_ALL_FALSE(nd::is_avail(NA - 1));
+  EXPECT_ALL_FALSE(nd::is_avail(NA * 1));
+  EXPECT_ALL_FALSE(nd::is_avail(NA / 1));
+
+  EXPECT_ALL_FALSE(nd::is_avail(1 + NA));
+  EXPECT_ALL_FALSE(nd::is_avail(1 - NA));
+  EXPECT_ALL_FALSE(nd::is_avail(1 * NA));
+  EXPECT_ALL_FALSE(nd::is_avail(1 / NA));
+
+  EXPECT_ALL_FALSE(nd::is_avail(NA + NA));
+  EXPECT_ALL_FALSE(nd::is_avail(NA - NA));
+  EXPECT_ALL_FALSE(nd::is_avail(NA * NA));
+  EXPECT_ALL_FALSE(nd::is_avail(NA / NA));
 }
 
```

Comment:
`EXPECT_FALSE` implicitly performs a static cast to bool by applying the logical not operator. Once a class overloads the logical not operator, the appropriate overload is called instead, so an array is returned from the expression rather than a boolean value. I didn't finish digging through the gtest code to find exactly why it fails after that. I could try fixing it in their code.